src/porepy/numerics/time_step_control.py

```diff
@@ -1,130 +1,209 @@
+            mssg_dtmin = "Initial time step cannot be smaller than minimum time step. "
+            mssg_dtmax = "Initial time step cannot be larger than maximum time step. "
+            mssg_unset = "This error was raised since `dt_min_max` was not set on "
+            mssg_unset += "initialization, and values of dt_min and dt_max were assigned "
+            mssg_unset += "based on the final simulation time. If you still want to use "
+            mssg_unset += "this initial time step, consider passing `dt_min_max` explictly."
+
+            if dt_init < dt_min_max[0]:
+                if dt_min_max_passed is not None:
+                    raise ValueError(mssg_dtmin)
+                else:
+                    raise ValueError(mssg_dtmin + mssg_unset)
+
+            if dt_init > dt_min_max[1]:
+                if dt_min_max_passed is not None:
+                    raise ValueError(mssg_dtmax)
+                else:
+                    raise ValueError(mssg_dtmax + mssg_unset)
+
+            # NOTE: The above checks guarantee that minimum time step <= maximum time step
+
+            # Sanity checks for maximum number of iterations.
+            # Note that 0 is a possibility. This will imply that the solver reaches
+            # convergence directly, e.g., as in linear solvers.
+            if iter_max < 0:
+                raise ValueError("Maximum number of iterations must be non-negative.")
+
+            # Sanity checks for optimal iteration range
+            if iter_optimal_range[0] > iter_optimal_range[1]:
+                s = "Lower optimal iteration range cannot be larger than"
+                s += " upper optimal iteration range."
+                raise ValueError(s)
+            elif iter_optimal_range[1] > iter_max:
+                s = "Upper optimal iteration range cannot be larger than"
+                s += " maximum number of iterations."
+                raise ValueError(s)
+            elif iter_optimal_range[0] < 0:
+                s = "Lower optimal iteration range cannot be negative."
+                raise ValueError(s)
+
+            # Sanity checks for lower and upper multiplication factors
+            if iter_lowupp_factor[0] <= 1:
+                raise ValueError("Expected lower multiplication factor > 1.")
+            elif iter_lowupp_factor[1] >= 1:
+                raise ValueError("Expected upper multiplication factor < 1.")
+
+            # Checks for sensible combinations of iter_optimal_range and iter_lowupp_factor
+            if dt_min_max[0] * iter_lowupp_factor[0] > dt_min_max[1]:
+                raise ValueError("Encountered dt_min * iter_low_factor > dt_max.")
+            elif dt_min_max[1] * iter_lowupp_factor[1] < dt_min_max[0]:
+                raise ValueError("Encountered dt_max * iter_upp_factor < dt_min.")
+
+            # Sanity check for recomputation factor
+            if recomp_factor >= 1:
+                raise ValueError("Expected recomputation multiplication factor < 1.")
```

Comment:
Again, warning

Reply:
In this case, we have to raise an error. Otherwise, the algorithm will (very easily) break.